src/verbs/queue_pair.rs

```diff
@@ -280,23 +297,20 @@ impl QueuePair for BasicQueuePair<'_> {
 
 #[derive(Debug)]
 pub struct ExtendedQueuePair<'res> {
-    // TODO: ibv_create_qp_ex returns ibv_qp instead of ibv_qp_ex, to be fixed
-    pub(crate) qp_ex: NonNull<ibv_qp>,
+    pub(crate) qp_ex: NonNull<ibv_qp_ex>,
     // phantom data for protection domain & completion queues
     _phantom: PhantomData<&'res ()>,
 }
 
 impl Drop for ExtendedQueuePair<'_> {
     fn drop(&mut self) {
-        // TODO: convert qp_ex to qp (port ibv_qp_ex_to_qp in rdma-mummy-sys)
         let ret = unsafe { ibv_destroy_qp(self.qp_ex.as_ptr().cast()) };
         assert_eq!(ret, 0)
     }
 }
 
 impl QueuePair for ExtendedQueuePair<'_> {
     unsafe fn qp(&self) -> NonNull<ibv_qp> {
-        // TODO: convert qp_ex to qp (port ibv_qp_ex_to_qp in rdma-mummy-sys)
         self.qp_ex.cast()
```

Comment:
Maybe using `NonNull::new_unchecked(&mut (*self.qp_ex.as_ptr()).qp_base as _)` to cast from `qp_ex` to `qp` is more portable, since this does not need to make any  assumption on the layout of the `struct ibv_qp_ex`.